### ISSUE TITLE
[Ansible/artifactory] Upgrade mode breaks existing installation

### ISSUE THREAD

**Issue by weakcamel** [jfrog/JFrog-Cloud-Installers#194]
**Is this a request for help?**:

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):

BUG REPORT

**Which installer**:

Ansible / Artifactory

**Which product and version** (eg: ansible & collection version - 7.24.2):

Ansible 5.0.1

Jfrog Platform collection 10.1.2

**Which operating system and version**(eg: ubuntu & version - 20.4):

Ubuntu Bionic 18.04

**What happened**:

In preparation for upgrade from Artifactory 7.19.8 to the latest, I bumped the jfrog collection version and tried to "preview" the changes by running playbook with "--check --diff" options.

* Artifactory 7.19.8
* Jfrog Platform Ansible Collection 7.19.8 -> 10.1.2

The result was quite surprising:
* firstly, role downloads the tarball with Artifactory (same version) and deflates it, stops Artifactory then it **deletes** existing installation which is already at the desired version (!).
* 2 tasks later role fails because it's missing a license file template

See log below.

**What you expected to happen**:

1. Running the role - no matter if in "upgrade" or "fresh install" mode should be idempotent, i.e. a no-op when running it version X -> X.  That's the basic promise of Ansible. Applicaton shouldn't be stopped, no files should be deleted etc.
2. Application files shouldn't IMO be deleted until the upgrade process ends and application is successfully up and running; it would be much safer and allow rollback if role deflated the application on the side and made a simple directory **move** to swap them. 
3.  Obviously, the failure in the last step should not happen (missing `artifactory.lic.j2` file) and upgrade should complete successfully

**How to reproduce it** (as minimally and precisely as possible):

1. Install Artifactory 7.19.8 using  7.19.8 collection
2. Upgrade to Jfrog Collection 10.1.2
3. Run Ansible against the same host in `--check --diff -e '{ "artifactory_upgrade_only": true }'` (see https://github.com/ansible/ansible/issues/17193 for boolean flags))

**Anything else we need to know**:

Log excerpt

```

PLAY [Protect against Jfrog Collection changes] ********************************

TASK [Gathering Facts] *********************************************************
ok: [artifactory.example.com]

PLAY [Prepare the zfs filesystem] **********************************************

TASK [Gathering Facts] *********************************************************
ok: [artifactory.example.com]

PLAY [Prepare system for Artifactory] ******************************************

TASK [Gathering Facts] *********************************************************
ok: [artifactory.example.com]

PLAY [Install and configure Artifactory] ***************************************

TASK [Gathering Facts] *********************************************************
ok: [artifactory.example.com]

TASK [Run the Jfrog Platform official roles to set up Artifactory] *************

TASK [jfrog.platform.artifactory : Perform installation] ***********************
skipping: [artifactory.example.com]

TASK [jfrog.platform.artifactory : Perform upgrade] ****************************
included: /Users/myuser/git/auto/galaxy-roles/ansible_collections/jfrog/platform/roles/artifactory/tasks/upgrade.yml for artifactory.example.com

TASK [jfrog.platform.artifactory : Check if artifactory tar already exists] ****
ok: [artifactory.example.com]

TASK [jfrog.platform.artifactory : Download artifactory for upgrade] ***********
changed: [artifactory.example.com]

TASK [jfrog.platform.artifactory : Extract artifactory tar] ********************
skipping: [artifactory.example.com]

TASK [jfrog.platform.artifactory : Stop artifactory] ***************************

TASK [jfrog.platform.artifactory : Ensure jfrog_home_directory exists] *********
ok: [artifactory.example.com]

TASK [jfrog.platform.artifactory : Delete artifactory app directory] ***********
--- before
+++ after
@@ -1,11800 +1,2 @@
 path: /opt/jfrog/artifactory/app
-path_content:
-  directories:
-  - /opt/jfrog/artifactory/app/misc
-  - /opt/jfrog/artifactory/app/frontend
-  - /opt/jfrog/artifactory/app/bin
-  - /opt/jfrog/artifactory/app/metadata
-  - /opt/jfrog/artifactory/app/artifactory
...

...
-  - /opt/jfrog/artifactory/app/event/bin/jf-event
-  - /opt/jfrog/artifactory/app/event/bin/event.sh
-state: directory
+state: absent

changed: [artifactory.example.com]

TASK [jfrog.platform.artifactory : Copy new app to artifactory app] ************
skipping: [artifactory.example.com]

TASK [jfrog.platform.artifactory : Configure single license] *******************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: If you are using a module and expect the file to exist on the remote, see the remote_src option
fatal: [artifactory.example.com]: FAILED! => changed=false 
  msg: |-
    Could not find or access 'artifactory.lic.j2'
    Searched in:
            /Users/myuser/git/auto/galaxy-roles/ansible_collections/jfrog/platform/roles/artifactory/templates/artifactory.lic.j2
            /Users/myuser/git/auto/galaxy-roles/ansible_collections/jfrog/platform/roles/artifactory/artifactory.lic.j2
            /Users/myuser/git/auto/galaxy-roles/ansible_collections/jfrog/platform/roles/artifactory/tasks/templates/artifactory.lic.j2
            /Users/myuser/git/auto/galaxy-roles/ansible_collections/jfrog/platform/roles/artifactory/tasks/artifactory.lic.j2
            /Users/myuser/git/auto/playbooks/templates/artifactory.lic.j2
            /Users/myuser/git/auto/playbooks/artifactory.lic.j2 on the Ansible Controller.
    If you are using a module and expect the file to exist on the remote, see the remote_src option

PLAY RECAP *********************************************************************
artifactory.example.com  : ok=9    changed=2    unreachable=0    failed=1    skipped=3    rescued=0    ignored=0   
```

**Comment by chukka** [comment]
@weakcamel  Thanks for reporting !  we removed `Configure single license` task in install flow and missed to remove in Upgrade  flow, will fix in next release (in few weeks)

**Comment by chukka** [comment]
Fixed in #200
